### PR TITLE
feat: Add agents filter in CSAT reports

### LINKED ISSUE #4039: User filter on CSAT reports
- [ ] Add user filter on `/api/v1/accounts/1/csat_survey_responses` API
- [ ] Allow multiple `user_ids` to be passed in the params
- [ ] Update the UI to select one or more users from the filter.

### PR DESCRIPTION
**Pull request by aswindevps**
## Description

This change allows the user to filter CSAT reports based on agents. Multiple agents can be selected in the filter.

Fixes #4039

## Type of change

- [ ] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

https://user-images.githubusercontent.com/5848740/156514066-6fef9ec2-6fe0-4e4d-b213-b3419713a4d5.mp4

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/controllers/api/v1/accounts/csat_survey_responses_controller.rb [hunk 1]
```diff
@@ -32,6 +32,7 @@ def set_csat_survey_responses
       Current.account.csat_survey_responses.includes([:conversation, :assigned_agent, :contact])
     )
     @csat_survey_responses = @csat_survey_responses.where(created_at: range) if range.present?
+    @csat_survey_responses = @csat_survey_responses.where(assigned_agent_id: params[:user_ids]) if params[:user_ids].present?
```
Comment (tejaswinichile) [inline app/controllers/api/v1/accounts/csat_survey_responses_controller.rb:35]: Can you check if you can add scopes something like below in the CSAT model, for range filter and agent filter  
```
scope :filter_by_created_at, lambda { |range| where(created_at: range) if range.present? }
```

Comment (aswindevps) [reply app/controllers/api/v1/accounts/csat_survey_responses_controller.rb]: Done

**Review by fayazara — APPROVED** [review]
Frontend code looks good. Tested on staging, working as expected.

**Review by sojan-official — APPROVED** [review]
LGTM

**Pull request merged**